File: backend/streetlight/src/main/java/com/kryeit/landmark/forum/petitions/Petition.java
package com.kryeit.landmark.forum.petitions;

import com.fasterxml.jackson.databind.node.ObjectNode;
import com.kryeit.landmark.LandmarkType;

import java.sql.Timestamp;
import java.util.HashMap;

public record Petition(
        long id, String description, long userId, LandmarkType type,
        double lat, double lon, HashMap<String, Boolean> info,
        Status status, Timestamp creation, Timestamp edition
) {

    public enum Status {
        PENDING,
        ACCEPTED,
        REJECTED
    }
}

File: frontend/landmarks/src/views/landmarks/js/Petition.js
import User from "@/js/auth/User.ts";

class Petition {
    constructor(id, description, userId, lat, lon, landmarkInfo, status, creation, edition, image) {
        this.id = id;
        this.description = description;
        this.userId = userId;
        this.lat = lat;
        this.lon = lon;
        this.landmarkInfo = landmarkInfo;
        this.status = status;
        this.creation = creation;
        this.edition = edition;
        this.image = image;
    }
}

export default Petition